clap-v3-utils/src/input_parsers/signer.rs

```diff
@@ -20,41 +18,6 @@ use {
 // Sentinel value used to indicate to write to screen instead of file
 pub const STDOUT_OUTFILE_TOKEN: &str = "-";
 
-// Return parsed values from matches at `name`
-pub fn values_of<T>(matches: &ArgMatches, name: &str) -> Option<Vec<T>>
-where
-    T: std::str::FromStr,
-    <T as std::str::FromStr>::Err: std::fmt::Debug,
-{
-    matches
-        .values_of(name)
-        .map(|xs| xs.map(|x| x.parse::<T>().unwrap()).collect())
-}
-
-// Return a parsed value from matches at `name`
-pub fn value_of<T>(matches: &ArgMatches, name: &str) -> Option<T>
-where
-    T: std::str::FromStr,
-    <T as std::str::FromStr>::Err: std::fmt::Debug,
-{
-    if let Some(value) = matches.value_of(name) {
-        value.parse::<T>().ok()
-    } else {
-        None
-    }
-}
-
-pub fn unix_timestamp_from_rfc3339_datetime(
-    matches: &ArgMatches,
-    name: &str,
-) -> Option<UnixTimestamp> {
-    matches.value_of(name).and_then(|value| {
-        DateTime::parse_from_rfc3339(value)
-            .ok()
-            .map(|date_time| date_time.timestamp())
-    })
-}
-
 // Return the keypair for an argument with filename `name` or None if not present.
 pub fn keypair_of(matches: &ArgMatches, name: &str) -> Option<Keypair> {
```

Comment:
I just want to double check if I am interpreting your suggestion correctly. I put the following in `input_parsers::mod.rs`
```
mod signer;
pub use signer::*;
```
This will make the refactored location `signer` private while re-exporting everything that was refactored into `signer` to be from `input_parsers` as before to make the refactor oblivious for the consumers.

Do you prefer to make `signer` a public module and let the consumers access it directly?

Reply:
i think that would expose too much. depending on whether we can sufficiently communicate the redirection to consumers with a single message, we can do something like this in `input_parsers/mod.rs`
```rust
#[deprecated("redirect message")]
pub use signers::{moved_symbol1, moved_symbol2, ...};
```

otherwise need a deprecation attribute and `pub use` for each moved symbol

```rust
#[deprecated("moved_symbol1 redirect message")]
pub use signers::moved_symbol1;

#[deprecated("moved_symbol2 redirect message")]
pub use signers::moved_symbol2;

...
```

Reply:
I apologize 🙏 , but I think I still need some clarification on the suggestion. All the functions in `signer` should not be deprecated (at least for this PR). They were simply moved to a separate module since the `input_parsers` module was getting a bit long. All the functions that are deprecated in this PR is in `input_validators` and just the single function `lamports_of_sol` in `input_parsers`. Given this, should I still add the following?
```rust
#[deprecated("redirect message")]
pub use signers::{...};
```

Reply:
what i proposed does not deprecate the symbols in `signer`, it deprecates _the old path to them_ in `input_parsers`.

a literal example. remember this code goes in `input_parsers/mod.rs`

```rust
#[deprecated(since = "1.17.0", note = "use `input_parsers::signer::keypair_of` instead")]
pub use signer::keypair_of;
```

this deprecates `input_parsers::keypair_of` in favor of `input_parsers::signer::keypair_of`.

now any consumer that previously had `use solana_clap_v3_utils::input_parsers::keypair_of` will get the deprecation warning suggesting that they update their use statement to point to `solana_clap_v3_utils::input_parsers::signer::keypair_of` instead. consumers referencing `keypair_of` from the `signer` submodule will not get the deprecation warning.

Reply:
Oh I see! Makes sense. I was thinking that I would just re-export the functions from their original location to just make the refactor oblivious for the consumers since it could be simpler if all parsers live in  `solana_clap_v3_utils::input_parsers` from the perspective of the consumers. If you prefer if we do a real divide of the paths to the two submodules in `input_parsers`, then I can definitely do that. I'll make the changes. I will probably need to update `keygen` and `zk-keygen` accordingly as well.

Reply:
Hm, after digging around for a while, it seems like there is no way to deprecate re-exports in rust yet unfortunately (https://github.com/rust-lang/rust/issues/30827) that triggers a deprecation warning. I'd be happy to simply leave the `input_parsers::signer` as a private module and let the consumers import from `input_parsers` directly as before. If you do feel strongly about deprecating the paths, then I can go ahead and deprecate them though they will not trigger any warnings for the consumers.

Reply:
@t-nelson, do you have any additional thoughts on this 🙏 ? I think we have two options:
- We can keep the original path for the `input_parsers::signer` as is without deprecating them. I don't really think there is anything unnatural about this, but interested to know what you think.
- We can make `input_parsers::signer` a public module and deprecate the old paths to them though the deprecation will not fire warnings if the old paths are used downstream.

Reply:
i don't understand. we already do this type of deprecation all over the code base.

Reply:
Yeah I realize that. My understanding is that all the existing deprecations of paths (not functions, types, etc, which are fine) throughout the codebase don't actually to trigger warnings even if the path is used by downstream. I quickly tried it out by using `solana_stake_program::id`, which should technically be deprecated (in favor of `solana_program::stake::program::id`), but doesn't seem to trigger warnings. I was really confused and found that https://github.com/rust-lang/rust/issues/30827 was still open.

For this PR, I'll just deprecate the `signer` as you suggested so that it is consistent with the rest of the codebase though 👍

Reply:
> ...don't actually to trigger warnings even if the path is used by downstream.

god dammit :melting_face: